contracts/Utils.sol

```diff
@@ -38,6 +42,10 @@ contract Utils {
 
     mapping(IERC20 => uint256) internal decimals;
 
+    /// @dev Sets the decimals of a token to storage if not already set, and returns
+    ///      the decimals value of the token
```

Comment:
please add:

prefer using this function over getDecimals - to avoid forgetting to set decimals in local storage